src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -355,6 +356,19 @@ public void edit() {
         });
     }
 
+    public void renameFile() {
+        LinkedFileRenameDialogView dialog = new LinkedFileRenameDialogView(this.linkedFile);
+        String oldFile = this.linkedFile.getLink();
+        int nameCount = Paths.get(oldFile).getNameCount();
+        Path subPath = Paths.get(oldFile).subpath(0, nameCount - 1);
+        Optional<LinkedFile> editedFile = dialog.showAndWait();
+        editedFile.ifPresent(file -> {
+            String newFile = System.getProperty("file.separator") + subPath.toString() + System.getProperty("file.separator") + file.getLink();
```

Comment:
You can use the [Path.resolveSibling](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#resolveSibling(java.lang.String)) method to construct the new path more easily.